pallets/connectors/src/lib.rs

```diff
@@ -177,14 +177,22 @@ pub mod pallet {
 			+ TypeInfo
 			+ MaxEncodedLen
 			+ Into<<Self as pallet_xcm_transactor::Config>::CurrencyId>
-			+ TryInto<
-				GeneralCurrencyIndex<u128, <Self as Config>::GeneralCurrencyPrefix>,
-				Error = DispatchError,
-			>;
+			+ TryInto<GeneralCurrencyIndexOf<Self>, Error = DispatchError>
+			+ TryFrom<GeneralCurrencyIndexOf<Self>, Error = DispatchError>
+			+ From<
+				<<<Self as Config>::ForeignInvestmentAccountant as InvestmentAccountant<
+					Self::AccountId,
+				>>::InvestmentInfo as InvestmentProperties<Self::AccountId>>::Currency,
+			> + CurrencyInspect<CurrencyId = <Self as pallet::Config>::CurrencyId>;
 
 		/// The converter from a DomainAddress to a Substrate AccountId
 		type AccountConverter: Convert<DomainAddress, Self::AccountId>;
 
+		/// The converter from a [ConnectorsWrappedCurrency] to `MultiLocation`.
+		type CurrencyConverter: Convert<ConnectorsWrappedCurrency, MultiLocation>
```

Comment:
Not sure about this name since this is not converting currencies. No suggestions now but something to think about

Reply:
How about `Wrapped{Currency/Token}Converter`? We have named `AccountConverter` based on the desired output type (i.e. a Substrate Account). Here, we use the converter for retrieving a `ConnectorsWrapped{Currency/Token}` from a `MultiLocation`.

Reply:
I like that :)